### PR TITLE
Open basedir language

### PR DESCRIPTION
**Pull request by escopecz**
## Description
If the fetching a language package fails for some reason, the user gets this untranslated message:

`mautic.core.could.not.set.language`

This PR describes each possible issue better and translates them. Also suggests a possible solution. The PR closes https://github.com/mautic/mautic/issues/1260.

Also reported in 
https://www.mautic.org/community/index.php/2707-languages-not-loaded
https://www.mautic.org/community/index.php/587-problem-setting-language
 
## Steps to reproduce

This will be the hard part. As Alan describes in the linked issue, this happens only if safe_mode or open_basedir are enabled. I tried to configure my server that way, but the languages are fetched successfully no matter how badly I try. So you can try to enable those in your php.ini and restart server. Hopefully, you'll be lucky. Michael said this issue happens on PHP 5.5. I'm using 5.6.4.

## Steps to test

That said, I hacked the code to fire different types of errors. I know it's a bad way to test it. If you could think of a better way, please share it.

The language package error messages can be triggered from 4 places:
1. While changing global language in the configuration.
2. While changing profile language in the profile form.
3. While editing a user.
4. While creating a new user.

Following tests should be done for each of the 4 places.

1. Add `$languageCode = 'blah';` to [this line](https://github.com/escopecz/mautic/blob/open-basedir-language/app/bundles/CoreBundle/Helper/LanguageHelper.php#L215). You should get error message about the unknown language.
2. Add `throw new \Exception('Hello world');` after [this line](https://github.com/escopecz/mautic/blob/open-basedir-language/app/bundles/CoreBundle/Helper/LanguageHelper.php#L231). You should get error message about the exception, which is "Hello world in this case".
3. Add `$data->code = 302` to [this line](https://github.com/escopecz/mautic/blob/open-basedir-language/app/bundles/CoreBundle/Helper/LanguageHelper.php#L244). You should get error message about the redirect issue with a link to the language package and description how to install it manually".
4. Add `$data->code = 404` to [this line](https://github.com/escopecz/mautic/blob/open-basedir-language/app/bundles/CoreBundle/Helper/LanguageHelper.php#L244). You should get error message about the language server problem and to try the language change later.
5. And of course, if you don't hack anything and the server allow PHP to follow redirects, then Mautic should be able to download and apply a language.

### CODE REVIEW

File: app/bundles/CoreBundle/Translations/en_US/flashes.ini [hunk 1]
```diff
@@ -8,3 +8,8 @@ mautic.core.error.notfound="Item was not found."
 mautic.core.notice.created="<a href='%url%' data-toggle='ajax' data-menu-link='%menu_link%'><strong>%name%</strong></a> has been created!"
 mautic.core.notice.deleted="<strong>%name%</strong> has been deleted!"
 mautic.core.notice.updated="<a href='%url%' data-toggle='ajax' data-menu-link='%menu_link%'><strong>%name%</strong></a> has been updated!"
+mautic.core.language.helper.error.fetching.package="An error occurred while downloading the language package."
+mautic.core.language.helper.error.follow.redirects="The configuration of your server doesn't allow Mautic to fetch the language package. Either safe_mode or open_basedir is turned on which doesn't allow CURL request to follow locations. You can download the language package <a href='%url%'>here</a>. Unzip it and upload it to the /translations directory. The language was reset to the default one for now."
```
Comment (alanhartless) [inline app/bundles/CoreBundle/Translations/en_US/flashes.ini]: Can we reword this to shorten it? I couldn't finish reading it before the flash disappeared.

**Comment by alanhartless** [comment]
Other than the note on the very long language string, this PR works for me 👍

File: app/bundles/CoreBundle/Translations/en_US/flashes.ini [hunk 2]
```diff
@@ -8,3 +8,8 @@ mautic.core.error.notfound="Item was not found."
 mautic.core.notice.created="<a href='%url%' data-toggle='ajax' data-menu-link='%menu_link%'><strong>%name%</strong></a> has been created!"
 mautic.core.notice.deleted="<strong>%name%</strong> has been deleted!"
 mautic.core.notice.updated="<a href='%url%' data-toggle='ajax' data-menu-link='%menu_link%'><strong>%name%</strong></a> has been updated!"
+mautic.core.language.helper.error.fetching.package="An error occurred while downloading the language package."
+mautic.core.language.helper.error.follow.redirects="Whoops, either safe_mode or open_basedir is turned on. Download the language from <a href='%url%'>here</a>. Unzip and upload it to the /translations directory."
+mautic.core.language.helper.invalid.language="Requested language '%language%' does not exist among the available Mautic Languages. The language was reset to the default one."
+mautic.core.language.helper.error.on.language.server.side="The language server responded with HTTP code %code%. It is probalby temporarily unavailable. Try to change the language later. The language was reset to the default one for now."
```
Comment (dbhurley) [inline app/bundles/CoreBundle/Translations/en_US/flashes.ini]: Typo on probably

File: app/bundles/CoreBundle/Translations/en_US/flashes.ini [hunk 3]
```diff
@@ -8,3 +8,8 @@ mautic.core.error.notfound="Item was not found."
 mautic.core.notice.created="<a href='%url%' data-toggle='ajax' data-menu-link='%menu_link%'><strong>%name%</strong></a> has been created!"
 mautic.core.notice.deleted="<strong>%name%</strong> has been deleted!"
 mautic.core.notice.updated="<a href='%url%' data-toggle='ajax' data-menu-link='%menu_link%'><strong>%name%</strong></a> has been updated!"
+mautic.core.language.helper.error.fetching.package="An error occurred while downloading the language package."
+mautic.core.language.helper.error.follow.redirects="Whoops, either safe_mode or open_basedir is turned on. Download the language from <a href='%url%'>here</a>. Unzip and upload it to the /translations directory."
+mautic.core.language.helper.invalid.language="Requested language '%language%' does not exist among the available Mautic Languages. The language was reset to the default one."
```
Comment (dbhurley) [inline app/bundles/CoreBundle/Translations/en_US/flashes.ini]: Language should not be capitalized

**Comment by dbhurley** [comment]
Left a couple of language comments. Otherwise +1 for me.
